### DISCUSSION TITLE
Confusing error text

### DISCUSSION THREAD

**Discussion by anton-kazlouski** [encode/django-rest-framework#7881] [General]
I'm not English native, but the error message seems to have a logical error:
`This field may not be null` -- means that a field might be null and might not be null (negative probability). But the translation means that the field is required and cannot be null. 

https://github.com/encode/django-rest-framework/blob/3578683a694e5414c363185b751bcb811d12358a/rest_framework/locale/ru_RU/LC_MESSAGES/django.po#L160

I've checked with Django sources and found this:
`This field cannot be null` which seems to me more accurate.

I want to clarify it's an actual mistake. If yes, I'll create a PR to fix this issue.

**Comment by lovelydinosaur** [comment]
Any of "cannot", "must not", "may not" would be okay here.
There's subtle differences in meaning, but they're mostly close enough to the same thing in this context.

Personally I probably marginally prefer "must not" or "may not", which implies "is not permitted", rather than cannot, which implies "is not possible".

I doubt we'll change it since it'd end up marking all the translations as needing re-translating (although they probably wouldn't actually), but I can see that determining *exactly* which phrasing would be most suitable for an international audience is a gray area.

**Reply by n2ygk** [reply]
Happily someone else already figured out the English ambiguity for us: https://tools.ietf.org/html/bcp14

**Reply by anton-kazlouski** [reply]
Originally, this question araised when I was reviwened a PR to the other projects. And found the following phrase:
`This field is required, may not be null or blank`

A developer said that he used DRF sources as a reference. Seems like, `This field is required` creates this contradiction.

**Reply by anton-kazlouski** [reply]
@tomchristie I won't push the idea of changing the entire project just to be as precise as possible. But could you, please, clarify this one moment, just to satisfy my sense of curiosity:
`Personally I probably marginally prefer "must not" or "may not", which implies "is not permitted"`

According to the link provided by @n2ygk and [Cambridge dictionary](https://dictionary.cambridge.org/ru/%D0%B3%D1%80%D0%B0%D0%BC%D0%BC%D0%B0%D1%82%D0%B8%D0%BA%D0%B0/%D0%B1%D1%80%D0%B8%D1%82%D0%B0%D0%BD%D1%81%D0%BA%D0%B0%D1%8F-%D0%B3%D1%80%D0%B0%D0%BC%D0%BC%D0%B0%D1%82%D0%B8%D0%BA%D0%B0/may)  may is not used in prohibition function.

**Reply by lovelydinosaur** [reply]
Sure.

![Screen Shot 2021-03-29 at 15 48 08](https://user-images.githubusercontent.com/647359/112854887-339f9780-90a6-11eb-837d-19ab7fe4aef7.png)

I can see some sense in technical writing prefering the more authorative "must not", because in a way "may" and "may not" do *not* exactly mean the logical inverse of each other.

"you may do this" means this "this is allowed". (But it's optional, you don't *have* to.)
"you may not do this" means "this is not allowed". (It doesn't mean, "this is optionally not allowed", which would be a nonsense.)

**Reply by anton-kazlouski** [reply]
I see now. Thanks!
